components/resources/library/src/skikoMain/kotlin/org/jetbrains/compose/resources/FontResources.skiko.kt

```diff
@@ -45,4 +49,26 @@ actual fun Font(resource: FontResource, weight: FontWeight, style: FontStyle): F
         }
     }
     return fontFile
+}
+
+@Composable
+actual fun Font(
+    resource: FontResource,
+    weight: FontWeight,
+    style: FontStyle,
+    variationSettings: FontVariation.Settings,
+): Font {
+    val resourceReader = LocalResourceReader.currentOrPreview
+    val fontFile by rememberResourceState(resource, weight, style, variationSettings, { defaultEmptyFont }) { env ->
+        val path = resource.getResourceItemByEnvironment(env).path
+        val settingsKey = variationSettings.settings
+            .sortedBy { it::class.simpleName }
+            .joinToString(",") { it::class.simpleName + "_" + it.hashCode() }
+        val key = "$path:$weight:$style:$settingsKey"
+        fontCache.getOrLoad(key) {
+            val fontBytes = resourceReader.read(path)
```

Comment:
Looks like this line creates a new ByteArray for every call. The fetch request will resolve from the cache, though. But still the new bytearray would require a new memory allocation (not only on web, but on all platforms).

A potential solution would be to add one more layer of caching - caching for raw bytes. 
But I'm not sure how often it would be needed. It would be very much useful when using the same variable font with many different variationSettings, or just different weight and style. 

 WDYT?

Reply:
It should be a rare operation in practice